packages/core/src/makeStyles.ts

```diff
@@ -1,15 +1,20 @@
 import { debugData, isDevToolsEnabled, getSourceURLfromError } from './devtools';
+import { insertionFactory } from './insertionFactory';
 import { resolveStyleRulesForSlots } from './resolveStyleRulesForSlots';
 import { reduceToClassNameForSlots } from './runtime/reduceToClassNameForSlots';
 import type { CSSClassesMapBySlot, CSSRulesByBucket, GriffelRenderer, StylesBySlots } from './types';
+import type { GriffelInsertionFactory } from './types';
 
 export interface MakeStylesOptions {
   dir: 'ltr' | 'rtl';
   renderer: GriffelRenderer;
 }
 
-export function makeStyles<Slots extends string | number>(stylesBySlots: StylesBySlots<Slots>) {
-  const insertionCache: Record<string, boolean> = {};
+export function makeStyles<Slots extends string | number>(
+  stylesBySlots: StylesBySlots<Slots>,
+  factory: GriffelInsertionFactory = insertionFactory,
+) {
+  const insertStyles = factory();
```

Comment:
would be nice to add tests for insertion factory usage

Reply:
I added tests for `insertionFactory()` itself and its connection with `makeStyles` 👍